parser-typechecker/src/Unison/Builtin/Terms.hs

```diff
@@ -19,11 +20,11 @@ import qualified Unison.Var as Var
 builtinTermsSrc :: Var v => a -> [(v, Term v a, Type v a)]
 builtinTermsSrc a =
   [ ( v "metadata.isPropagated",
-      Term.constructor a Decls.isPropagatedRef Decls.isPropagatedConstructorId,
+      Term.constructor a (ConstructorReference Decls.isPropagatedRef Decls.isPropagatedConstructorId),
```

Comment:
I think that would be going in the opposite direction of this PR, which is to give a name to `(Reference, Int)` rather than pass them around as separate arguments everywhere.

But I think we could go further here, and have a top-level 

```haskell
Decls.isPropagatedConstructor = ConstructorReference Decls.isPropagatedRef Decls.isPropagatedConstructorId
```

for this builtin, and all others, whose refs/conIds we refer to explicitly